simple-vm/README.md

```diff
@@ -0,0 +1,30 @@
+# Demo deployment on Linux VM
+
+## Goal :rocket:
+
+- [x] To create the target infrastructure using Terraform
+- [x] The configuration of the server and application will be done via Ansible
+- [x] To setup DNS for the target using Terraform
+
+## Details :memo:
+
+The applications that will be deployed
+1. **Go**: [A minimal golang application](https://github.com/PhilomathesInc/demo-applications/tree/main/minimal-go-app)
+1. **PHP**: [A minimal PHP application](https://github.com/PhilomathesInc/demo-applications/tree/main/minimal-php-app)
+
+## Steps
+
+### Terraform
+1. [Install Terraform](https://learn.hashicorp.com/tutorials/terraform/install-cli)
+1. Run `terraform init`
+1. Generate a SSH key pair - `ssh-keygen -t ed25519 -C "$(whoami)@$(uname -n)-$(gdate -I)" -N '' -f id_ed25519`
```

Comment:
Is it required to generate a new SSH key pair? If not, you could mention how to use existing SSH key.

Reply:
It is definitely a best-practice to always use new keys and not reuse existing keys. 

Let's say 
1. two developers need access to the same _dev_ machine, they should have two separate keys. 
2. In another situation - a single developer needs access to two machines - _dev_ and _prod_, they should actually have two separate keys.

In the real world, for an organization, we'd use a tool like Vault to manage SSH keys and rotate them or issue short term SSH keys per user/per machine.

---
**_tl;dr_ - I don't think we should reuse existing SSH keys.**

Reply:
Does it make sense to create the SSH keys using terraform itself using [tls_private_key](https://registry.terraform.io/providers/hashicorp/tls/latest/docs/resources/private_key)?

Reply:
Not according to the docs themselves, (from the same link you mentioned)

> The private key generated by this resource will be stored unencrypted in your Terraform state file. Use of this resource for production deployments is not recommended. Instead, generate a private key file outside of Terraform and distribute it securely to the system where Terraform will be run.